mockingbird/src/commonMain/kotlin/com/careem/mockingbird/test/Functions.kt

```diff
@@ -192,20 +199,25 @@ public fun <T : Spy, R> T.spy(
     arguments: Map<String, Any?> = emptyMap(),
     delegate: () -> R
 ): R {
-    val hashCode = this.hashCode()
+    val uuid = this.uuid
     return MockingBird.invocationRecorder().access { recorder ->
         val invocation = Invocation(methodName = methodName, arguments = arguments)
-        recordInvocation(hashCode, recorder, invocation)// TODO change name
+        recordInvocation(uuid, recorder, invocation)// TODO change name
         @Suppress("UNCHECKED_CAST")
         val mockResponse = recorder.getResponse(
-            instanceHash = hashCode,
+            uuid = uuid,
             invocation = invocation,
             relaxed = true
         ) as R
         return@access mockResponse ?: delegate()
     }
 }
 
+
+public fun <T : Mock> T.uuid(): Lazy<String> {
+    return lazy { "${this.hashCode()}-${uuidGenerator.getAndAdd(1)}" }
```

Comment:
Using UUIDs as keys ought to be enough.

Reply:
Yes it is, My though here was to keep the hashcode to protect ourself from bad clients, for bad clients I mean clients that do not override the `uuid` in the right way, do you think I am overcomplicating it?

Reply:
Good call actually.